src/util/records.py

```diff
@@ -409,7 +409,13 @@ def remove_records(cr, model, ids):
                 remove_records(cr, inh.model, [rid for (rid,) in cr.fetchall()])
 
     table = table_of_model(cr, model)
-    cr.execute('DELETE FROM "{}" WHERE id IN %s'.format(table), [ids])
+    parallel_execute(
+        cr,
+        [
+            cr.mogrify(format_query(cr, "DELETE FROM {} WHERE id IN %s", table), [chunk_ids]).decode()
+            for chunk_ids in chunks(ids, 1000, fmt=tuple)
+        ],
+    )
```

Comment:
np to not format the query many times:

```suggestion
    query = format_query(cr, "DELETE FROM {} WHERE id IN %s", table)
    parallel_execute(
        cr,
        [
            cr.mogrify(query, [chunk_ids]).decode()
            for chunk_ids in chunks(ids, chunk_size, fmt=tuple)
        ],
    )
```

Also, I'd argue for a `chunk_size` parameter.